peripherals/i2c/i2c_e1.yaml

```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 The gd32-rs authors.
+#
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+_include:
+  - i2c_common.yaml
+  - i2c_high.yaml
+
+"I2C*":
+  SADDR0:
+    ADDRESS0: [0, 1]
+    ADDRESS7_1: [0, 127]
+    ADDRESS9_8: [0, 3]
```

Comment:
You might want to merge these into a single field, the same way as `common_patches/gd32f30x.yaml` does.

Reply:
Hmm, that's true. I'd be inclined to handle that in the HAL though, it's easier to do a right shift in the 7-bit addressing case than to break the 10 bit address into three parts, IMO. It's also consistent with the other PAC crates here to treat it as a single field.

Reply:
Done, I also added a comment in the addressing mode enum to make sure we know what to do when using 7b addresses :)